auth/middleware.go

```diff
@@ -77,6 +73,10 @@ func (k *kratosMiddleware) validateSession(r *http.Request) (*client.Session, er
 		return &login.Session, nil
 	}
 
+	if strings.HasPrefix(r.URL.Path, "/upstream_push") && !hasBasicAuth {
+		return nil, errors.New("endpoint requires basic auth")
+	}
```

Comment:
@moshloop should I remove this? So upstream can also work with cookies?

Reply:
Yeh - I don't think we should different auth methods for different endpoints